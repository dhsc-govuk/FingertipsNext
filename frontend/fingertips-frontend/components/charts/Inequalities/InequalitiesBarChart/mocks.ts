import { InequalitiesBarChartData } from '@/components/charts/Inequalities/helpers/inequalitiesHelpers';

export const getTestData = (): InequalitiesBarChartData => ({
  areaCode: 'A1425',
  areaName: 'South FooBar',
  data: {
    period: 2008,
    inequalities: {
      Male: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: false,
      },
      Persons: {
        value: 135.149304,
        count: 222,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: true,
      },
      Female: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
        isAggregate: false,
      },
    },
  },
});
