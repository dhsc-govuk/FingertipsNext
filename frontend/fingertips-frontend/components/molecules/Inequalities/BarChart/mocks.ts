import { InequalitiesBarChartData } from '@/components/organisms/Inequalities/inequalitiesHelpers';

export const getTestData = (): InequalitiesBarChartData => ({
  areaName: 'South FooBar',
  data: {
    period: 2008,
    inequalities: {
      Male: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
      },
      Persons: {
        value: 135.149304,
        count: 222,
        upper: 578.32766,
        lower: 441.69151,
      },
      Female: {
        value: 890.328253,
        count: 131,
        upper: 578.32766,
        lower: 441.69151,
      },
    },
  },
});
