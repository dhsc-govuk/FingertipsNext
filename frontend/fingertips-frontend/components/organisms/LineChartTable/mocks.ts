export const MOCK_ENGLAND_DATA = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      year: 2004,
      count: 200,
      value: 904.874,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '0-4',
      sex: 'Female',
    },
    {
      year: 2008,
      count: 500,
      value: 965.9843,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '10-14',
      sex: 'Female',
    },
  ],
};

export const MOCK_HEALTH_DATA = [
  {
    areaCode: 'A1425',
    areaName: 'Greater Manchester ICB - 00T',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 890.305692,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
    ],
  },
  {
    ...MOCK_ENGLAND_DATA,
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        year: 2004,
        count: 222,
        value: 135.149304,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Persons',
      },
    ],
  },
];
