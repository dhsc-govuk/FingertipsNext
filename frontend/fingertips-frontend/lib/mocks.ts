import {
  AgeData,
  DeprivationData,
  HealthDataPoint,
  HealthDataPointTrendEnum,
  SexData,
} from '@/generated-sources/ft-api-client';
import { mockDatePeriod } from '@/mock/data/mockDatePeriod';

export const noDeprivation: DeprivationData = {
  sequence: 1,
  type: 'All',
  value: 'All',
  isAggregate: true,
};

export const personsSex: SexData = {
  value: 'Persons',
  isAggregate: true,
};

export const maleSex: SexData = {
  value: 'Male',
  isAggregate: false,
};

export const femaleSex: SexData = {
  value: 'Female',
  isAggregate: false,
};

export const allAgesAge: AgeData = {
  value: 'All ages',
  isAggregate: true,
};

export const disaggregatedAge = (age: string) => {
  return {
    value: age,
    isAggregate: false,
  };
};

export const healthDataPoint: HealthDataPoint = {
  count: 389,
  lowerCi: 441.69151,
  upperCi: 578.32766,
  value: 278.29134,
  datePeriod: mockDatePeriod(2006),
  sex: personsSex,
  ageBand: allAgesAge,
  trend: HealthDataPointTrendEnum.NotYetCalculated,
  deprivation: noDeprivation,
};
