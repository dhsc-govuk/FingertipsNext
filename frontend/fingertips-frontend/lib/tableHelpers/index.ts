export enum LineChartTableHeadingEnum {
  AreaPeriod = 'Period',
  BenchmarkTrend = 'Compared to benchmark',
  AreaCount = 'Count',
  AreaValue = 'Value',
  AreaLower = 'Lower',
  AreaUpper = 'Upper',
  BenchmarkValue = 'Value ',
}

export enum InequalitiesSexTableHeadingsEnum {
  PERIOD = 'Period',
  PERSONS = 'Persons',
  MALE = 'Male',
  FEMALE = 'Female',
  BENCHMARK = 'England benchmark',
}

export interface LineChartTableRowData {
  period: number;
  count: number;
  value: number;
  lower: number;
  upper: number;
}

export interface InequalitiesSexTableRowData {
  period: number;
  persons: number | undefined;
  male: number | undefined;
  female: number | undefined;
  englandBenchmark: number | undefined;
}

export const convertToPercentage = (value: number): string => {
  // dummy function to do percentage conversions until real conversion logic is provided
  return `${((value / 10000) * 100).toFixed(1)}%`;
};
