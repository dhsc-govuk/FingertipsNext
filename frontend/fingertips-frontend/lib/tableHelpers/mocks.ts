export const MOCK_ENGLAND_DATA = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      year: 2004,
      count: 200,
      value: 904.874,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '0-4',
      sex: 'All',
    },
    {
      year: 2008,
      count: 500,
      value: 965.9843,
      lowerCi: 0,
      upperCi: 0,
      ageBand: '10-14',
      sex: 'All',
    },
  ],
};

export const MOCK_HEALTH_DATA = [
  {
    areaCode: 'A1425',
    areaName: 'Greater Manchester ICB - 00T',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 890.305692,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Male',
      },
      {
        year: 2004,
        count: 267,
        value: 703.420759,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Female',
      },
    ],
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        year: 2008,
        count: 222,
        value: 135.149304,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'All',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Male',
      },
      {
        year: 2008,
        count: 131,
        value: 890.328253,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        ageBand: 'All',
        sex: 'Female',
      },
    ],
  },
];

export const MOCK_PARENT_DATA = {
  areaCode: 'E40000014',
  areaName: 'NHS North West Region',
  healthData: [
    {
      year: 2008,
      count: 777,
      value: 1000,
      lowerCi: 500,
      upperCi: 1500,
      ageBand: 'All',
      sex: 'All',
    },
    {
      year: 2004,
      count: 777,
      value: 1000,
      lowerCi: 500,
      upperCi: 1500,
      ageBand: 'All',
      sex: 'All',
    },
  ],
};
