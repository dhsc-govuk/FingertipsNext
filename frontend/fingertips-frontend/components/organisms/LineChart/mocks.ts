import {
  HealthDataForArea,
  HealthDataPointTrendEnum,
} from '@/generated-sources/ft-api-client';
import { allAgesAge, noDeprivation, personsSex } from '@/lib/mocks';

export const mockIndicatorData: HealthDataForArea[] = [
  {
    areaCode: 'A1425',
    areaName: 'North FooBar',
    healthData: [
      {
        count: 389,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        value: 278.29134,
        year: 2006,
        periodLabel: '2006',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 267,
        lowerCi: 441.69151,
        upperCi: 578.32766,
        value: 703.420759,
        year: 2004,
        periodLabel: '2004',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
  {
    areaCode: 'A1426',
    areaName: 'South FooBar',
    healthData: [
      {
        count: 654,
        lowerCi: 750.69151,
        upperCi: 800.32766,
        value: 786.27434,
        year: 2010,
        periodLabel: '2010',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 567,
        lowerCi: 440.69151,
        upperCi: 420.32766,
        value: 435.420759,
        year: 2007,
        periodLabel: '2007',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
  {
    areaCode: 'A1427',
    areaName: 'East FooBar',
    healthData: [
      {
        count: 953,
        lowerCi: 460.69151,
        upperCi: 500.32766,
        value: 478.27434,
        year: 2020,
        periodLabel: '2020',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
      {
        count: 567,
        lowerCi: 220.69151,
        upperCi: 250.32766,
        value: 234.420759,
        year: 2012,
        periodLabel: '2012',
        sex: personsSex,
        ageBand: allAgesAge,
        trend: HealthDataPointTrendEnum.NotYetCalculated,
        deprivation: noDeprivation,
        isAggregate: true,
      },
    ],
  },
];

export const mockBenchmarkData: HealthDataForArea = {
  areaCode: 'E92000001',
  areaName: 'England',
  healthData: [
    {
      count: 389,
      lowerCi: 441.69151,
      upperCi: 578.32766,
      value: 278.29134,
      year: 2006,
      periodLabel: '2006',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
    {
      count: 267,
      lowerCi: 441.69151,
      upperCi: 578.32766,
      value: 703.420759,
      year: 2004,
      periodLabel: '2004',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
  ],
};

export const mockParentData: HealthDataForArea = {
  areaCode: 'P001',
  areaName: 'Parent',
  healthData: [
    {
      count: 100,
      lowerCi: 200,
      upperCi: 400,
      value: 300,
      year: 2006,
      periodLabel: '2006',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
    {
      count: 101,
      lowerCi: 201,
      upperCi: 401,
      value: 301,
      year: 2004,
      periodLabel: '2004',
      sex: personsSex,
      ageBand: allAgesAge,
      trend: HealthDataPointTrendEnum.NotYetCalculated,
      deprivation: noDeprivation,
      isAggregate: true,
    },
  ],
};
