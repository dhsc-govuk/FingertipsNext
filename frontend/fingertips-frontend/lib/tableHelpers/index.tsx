import { Table } from 'govuk-react';
import styled from 'styled-components';
import { typography } from '@govuk-react/lib';
import {
  HealthDataForArea,
  HealthDataPoint,
} from '@/generated-sources/ft-api-client';
import { Sex } from '@/components/organisms/Inequalities/inequalitiesHelpers';
import { GovukColours } from '../styleHelpers/colours';

export enum LineChartTableHeadingEnum {
  AreaPeriod = 'Period',
  BenchmarkTrend = 'Compared to benchmark',
  AreaCount = 'Count',
  AreaValue = 'Value',
  AreaLower = 'Lower',
  AreaUpper = 'Upper',
  BenchmarkValue = 'Value ',
}

export enum InequalitiesSexTableHeadingsEnum {
  PERIOD = 'Period',
  PERSONS = 'Persons',
  MALE = 'Male',
  FEMALE = 'Female',
  BENCHMARK = 'England benchmark',
}

export interface LineChartTableRowData {
  period: number;
  count?: number;
  value?: number;
  lower?: number;
  upper?: number;
}

export interface InequalitiesSexTableRowData {
  period: number;
  persons?: number;
  male?: number;
  female?: number;
  englandBenchmark?: number;
}

export const StyledTableCellHeader = styled(Table.CellHeader)(
  typography.font({ size: 14 }),
  {
    fontWeight: 'bold',
    padding: '0.625em 0',
  }
);

export const StyledAlignRightHeader = styled(StyledTableCellHeader)({
  textAlign: 'right',
  paddingRight: '10px',
  verticalAlign: 'top',
});

export const StyledGreyHeader = styled(StyledAlignRightHeader)({
  backgroundColor: GovukColours.MidGrey,
  borderTop: `solid #F3F2F1 2px`,
  width: '16%',
});

export const StyledTableCell = styled(Table.Cell)(
  typography.font({ size: 14 }),
  {
    paddingRight: '0',
  }
);

export const StyledAlignRightTableCell = styled(StyledTableCell)({
  textAlign: 'right',
  paddingRight: '10px',
});

export const StyledGreyTableCellValue = styled(StyledAlignRightTableCell)({
  backgroundColor: GovukColours.MidGrey,
  borderTop: `solid #F3F2F1 2px`,
});

export const StyledAlignLeftTableCell = styled(StyledTableCell)({
  textAlign: 'left',
  width: '10%',
});

export const StyledAlignLeftHeader = styled(StyledTableCellHeader)({
  textAlign: 'left',
  verticalAlign: 'top',
});

export const StyledDiv = styled('div')({
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
});

// When value is undefined, it returns an X with an aria-label for screen readers.
export const convertToPercentage = (value?: number): React.ReactNode => {
  if (value === undefined) {
    return (
      <span aria-label="Not available" data-testid="not-available">
        <span aria-hidden="true">X</span>
      </span>
    );
  }
  return `${((value / 10000) * 100).toFixed(1)}%`;
};

export const mapToLineChartTableData = (
  areaData: HealthDataForArea
): LineChartTableRowData[] =>
  areaData.healthData.map((healthPoint) => ({
    period: healthPoint.year,
    count: healthPoint.count,
    value: healthPoint.value,
    lower: healthPoint.lowerCi,
    upper: healthPoint.upperCi,
  }));

export const mapToInequalitiesSexTableData = (
  groupedYearData: Record<number, HealthDataPoint[] | undefined>,
  englandBenchmarkData: Record<number, HealthDataPoint[] | undefined>
): InequalitiesSexTableRowData[] => {
  return Object.keys(groupedYearData).map((key) => ({
    period: Number(key),
    persons: groupedYearData[Number(key)]?.find((data) => data.sex === Sex.ALL)
      ?.value,
    male: groupedYearData[Number(key)]?.find((data) => data.sex === Sex.MALE)
      ?.value,
    female: groupedYearData[Number(key)]?.find(
      (data) => data.sex === Sex.FEMALE
    )?.value,
    englandBenchmark: englandBenchmarkData[Number(key)]?.find(
      (data) => data.sex === Sex.ALL
    )?.value,
  }));
};
