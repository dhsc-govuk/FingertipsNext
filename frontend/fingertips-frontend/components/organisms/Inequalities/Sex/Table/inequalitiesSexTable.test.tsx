import { render, screen } from '@testing-library/react';
import { expect } from '@jest/globals';
import { InequalitiesSexTable } from '.';
import { MOCK_ENGLAND_DATA, MOCK_HEALTH_DATA } from '@/lib/tableHelpers/mocks';
import { InequalitiesSexTableHeadingsEnum } from '@/lib/tableHelpers';

describe('Inequalities sex table suite', () => {
  const CELLS_PER_ROW = 5;
  it('should render the InequalitiesSexTable component', () => {
    render(
      <InequalitiesSexTable
        healthIndicatorData={MOCK_HEALTH_DATA[1]}
        englandBenchmarkData={MOCK_ENGLAND_DATA}
      />
    );

    expect(
      screen.getByTestId('inequalitiesSexTable-component')
    ).toBeInTheDocument();
  });

  it('should render expected elements', () => {
    render(
      <InequalitiesSexTable
        healthIndicatorData={MOCK_HEALTH_DATA[1]}
        englandBenchmarkData={MOCK_ENGLAND_DATA}
      />
    );

    expect(screen.getByRole('table')).toBeInTheDocument();
    expect(screen.getAllByRole('columnheader')[0]).toHaveTextContent(
      MOCK_HEALTH_DATA[1].areaName
    );
    expect(screen.getAllByRole('cell')).toHaveLength(CELLS_PER_ROW);
    Object.values(InequalitiesSexTableHeadingsEnum).forEach((heading, index) =>
      expect(
        screen.getByTestId(`header-${heading}-${index}`)
      ).toBeInTheDocument()
    );
  });

  it('should display x if data point is not available', () => {
    const expectedNumberOfRows = 2;
    render(
      <InequalitiesSexTable
        healthIndicatorData={MOCK_HEALTH_DATA[0]}
        englandBenchmarkData={MOCK_ENGLAND_DATA}
      />
    );
    expect(screen.getAllByRole('cell')).toHaveLength(
      expectedNumberOfRows * CELLS_PER_ROW
    );
    screen
      .getAllByRole('paragraph')
      .forEach((paragraph) => expect(paragraph).toHaveTextContent('x'));
  });

  it('snapshot test - should match snapshot', () => {
    const container = render(
      <InequalitiesSexTable
        healthIndicatorData={MOCK_HEALTH_DATA[1]}
        englandBenchmarkData={MOCK_ENGLAND_DATA}
      />
    );
    expect(container.asFragment()).toMatchSnapshot();
  });
});
