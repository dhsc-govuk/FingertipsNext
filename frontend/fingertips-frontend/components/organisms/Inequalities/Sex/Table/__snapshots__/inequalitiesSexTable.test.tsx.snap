// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inequalities sex table suite snapshot test - should match snapshot 1`] = `
<DocumentFragment>
  .c2 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
  font-weight: 700;
}

.c2:last-child {
  padding-right: 0;
}

.c9 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
}

.c9:last-child {
  padding-right: 0;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  margin-bottom: 20px;
}

.c5 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 12px;
  line-height: 1.25;
  font-weight: bold;
  padding: 0.625em 0;
}

.c6 {
  text-align: right;
  padding-right: 10px;
  vertical-align: top;
}

.c7 {
  background-color: #B1B4B6;
  border-top: solid #F3F2F1 2px;
  width: 16%;
}

.c10 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 12px;
  line-height: 1.25;
  padding-right: 0;
}

.c14 {
  text-align: right;
  padding-right: 10px;
}

.c15 {
  background-color: #B1B4B6;
  border-top: solid #F3F2F1 2px;
}

.c11 {
  text-align: left;
  width: 10%;
}

.c8 {
  text-align: left;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 12px;
  line-height: 1.25;
  font-weight: bold;
  padding: 0.625em 0;
}

.c12 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 12px;
  line-height: 1.25;
}

.c4 {
  text-align: center;
}

.c13 {
  text-align: right;
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 30px;
  }
}

@media print {
  .c5 {
    font-size: 12px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c5 {
    font-size: 14px;
    line-height: 1.4285714285714286;
  }
}

@media print {
  .c10 {
    font-size: 12px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c10 {
    font-size: 14px;
    line-height: 1.4285714285714286;
  }
}

@media print {
  .c3 {
    font-size: 12px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c3 {
    font-size: 14px;
    line-height: 1.4285714285714286;
  }
}

@media print {
  .c12 {
    font-size: 12px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c12 {
    font-size: 14px;
    line-height: 1.4285714285714286;
  }
}

<div
    class="c0"
    data-testid="inequalitiesSexTable-component"
  >
    <table
      class="c1"
    >
      <thead
        class=""
      >
        <tr
          class=""
        >
          <th
            class="c2 c3 c4"
            colspan="4"
          >
            South FooBar
          </th>
          <th
            class="c2 c5 c6 c7"
          />
        </tr>
        <tr
          class=""
        >
          <th
            class="c2 c5 c8"
            data-testid="header-Period-0"
          >
            Period
          </th>
          <th
            class="c2 c5 c6"
            data-testid="header-Persons-1"
          >
            Persons
          </th>
          <th
            class="c2 c5 c6"
            data-testid="header-Male-2"
          >
            Male
          </th>
          <th
            class="c2 c5 c6"
            data-testid="header-Female-3"
          >
            Female
          </th>
          <th
            class="c2 c5 c6 c7"
            data-testid="header-England benchmark-4"
          >
            England benchmark
          </th>
        </tr>
      </thead>
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <td
            class="c9 c10 c11"
          >
            2008
          </td>
          <td
            class="c9 c12 c13"
          >
            1.4%
          </td>
          <td
            class="c9 c12 c13"
          >
            8.9%
          </td>
          <td
            class="c9 c12 c13"
          >
            8.9%
          </td>
          <td
            class="c9 c10 c14 c15"
          >
            9.7%
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
